src/kernel/boot.c

```diff
@@ -580,19 +580,22 @@ BOOT_CODE static bool_t provide_untyped_cap(
     cap_t      root_cnode_cap,
     bool_t     device_memory,
     pptr_t     pptr,
-    word_t     size_bits)
+    word_t     size_bits,
+    seL4_SlotRegion *slot_region
```

Comment:
It's generally preferred to use global variables where possible in these functions. Is there a reason that this needs to be changed?

Reply:
In this case the proofs are fine with the change.  Also I forgot that slot_region is just a fixed size struct and not an array.